MdeModulePkg/Library/ArmFfaLib/ArmFfaPeiLib.c

```diff
@@ -74,6 +138,15 @@ ArmFfaPeiLibConstructor (
     return Status;
   }
 
+  Status = (*PeiServices)->LocatePpi (
+                             PeiServices,
+                             &gEfiPeiMemoryDiscoveredPpiGuid,
+                             0,
+                             NULL,
+                             &Dummy
+                             );
+  TempMem = (EFI_ERROR (Status)) ? TRUE : FALSE;
+
   RxTxBufferHob = GetFirstGuidHob (&gArmFfaRxTxBufferInfoGuid);

```

Comment:
This seems to be 85% duplicated functionality as in `PeiServicesMemoryDiscoveredNotifyCallback`? Can we please define a common routine for both places?